lib/flight_scheduler/commands/info.rb

```diff
@@ -28,43 +28,144 @@
 module FlightScheduler
   module Commands
     class Info < Command
-      extend OutputMode::TLDR::Index
+      class Lister
+        include OutputMode::TLDR::Index
 
-      # Wraps the partition object after it's nodes have been filtered by a state
+        # Returns true if the output will contain state information, else false
+        def state?
+          @state ? true : false
+        end
+
+        # Returns true if the output will contain each hostname individually, else false
+        def hostnames?
+          @hostnames ? true : false
+        end
+
+        def register_default_columns
+          register_partition
+          register_column(header: 'AVAIL') { |_| 'TBD' }
+          register_column(header: 'TIMELIMIT') { |_| 'TBD' }
+          register_nodes
+          register_state
+          register_nodelist
+        end
+
+        def register_partition
+          register_column(header: 'PARTITION') { |p| p.name }
+        end
+
+        def register_nodes
+          register_column(header: 'NODES') { |p| p.nodes.length }
+        end
+
+        def register_state
+          @state = true
+          register_column(header: 'STATE') { |p| p.state }
+        end
+
+        def register_nodelist
+          register_column(header: 'NODELIST') { |p| p.nodes.map(&:name).join(',') }
+        end
+
+        # NOTE: This method is used to quasi-list the nodes instead of partitions
+        #
+        # It assumes a one-to-one mapping of partitions to nodes. This is achieved using
+        # the PartitionProxy and does not represent the underlining data model
+        def register_hostnames
+          @hostnames = true
+          register_column(header: 'HOSTNAMES') { |p| p.nodes.first.name }
+        end
+
+        def register_cpus
+          register_column(header: 'CPUS') do |p|
+            p.nodes.reduce(0) { |sum, n| sum += n.cpus.to_i }
```

Comment:
This should be the number of CPUs per node, not the total number of CPUs.  If the nodes have differing numbers of CPUs it should output "<minimum CPUs>+".  Here is an example from slurm:

```
[flight@gateway1 (ben2) ~]$ grep NodeName /opt/flight/opt/slurm/etc/slurm.conf
NodeName=node01 CPUs=1 RealMemory=1024
NodeName=node02 CPUs=2 RealMemory=2048
[flight@gateway1 (ben2) ~]$ sinfo  --Format NodeHost,CPUs,Memory
HOSTNAMES           CPUS                MEMORY              
node01              1                   1024                
node02              2                   2048                
[flight@gateway1 (ben2) ~]$ sinfo  --Format NodeList,CPUs,Memory
NODELIST            CPUS                MEMORY              
node[01-02]         1+                  1024+               
[flight@gateway1 (ben2) ~]$ 

```